spec/integration/precompilation_spec.rb

```diff
@@ -43,4 +43,26 @@
     sprite_files = Dir.glob(File.join('public', 'assets', 'sprites', 'application-*.png'))
     expect(sprite_files.length).to eq(2)
   end
+
+  context 'a file is renamed' do
+    before do
+      compile_assets
+      FileUtils.mv 'app/assets/images/application/fool.png', 'app/assets/images/application/ffool.png'
+      compile_assets
```

Comment:
I expect a new sprite to be generated, because the order of the files in the sprite has changed. Without the change in `Collection#cache_key`, a new sprite is not generated (which is why the spec on line 56 below expects 2 files).